bayes_kit/iat.py

```diff
@@ -29,60 +32,56 @@ def _end_pos_pairs(chain: VectorType) -> IntType:
     ```
 
     Parameters:
-    chain (VectorType): Input sequence (typically of autocorrelations).
+    acor (VectorType): Input sequence of autocorrelations at lag given by index.
 
     Returns:
     The index 1 past the last positive pair of values starting on an even index.
     """
-    N = len(chain)
+    N = len(acor)
     n = 0
     while n + 1 < N:
-        if chain[n] + chain[n + 1] < 0:
+        if acor[n] + acor[n + 1] < 0:
             return n
         n += 2
     return n
 
 
 def iat_ipse(chain: VectorType) -> FloatType:
     """
-    Return an estimate of the integrated autocorrelation time (IAT) 
+    Return an estimate of the integrated autocorrelation time (IAT)
     of the specified Markov chain using the initial positive sequence
     estimator (IPSE).
 
     The integrated autocorrelation time of a chain is defined to be
     the sum of the autocorrelations at every lag (positive and negative).
-    If `ac[n]` is the autocorrelation at lag `n`, then 
+    If `autocorr[n]` is the autocorrelation at lag `n`, then
 
     ```
-    IAT = SUM_{n in Z} ac[n],
+    IAT = SUM_{n in Z} autocorr[n],
     ```
-    
-    where `Z = {..., -2, -1, 0, 1, 2, ...}` is the set of integers.
 
+    where `Z = {..., -2, -1, 0, 1, 2, ...}` is the set of integers.
 
-    Because the autocorrelations are symmetric, `ac[n] == ac[-n]` and
-    `ac[0] = 1`, this reduces to
+    Because the autocorrelations are symmetric, `autocorr[n] == autocorr[-n]` and
+    `autocorr[0] = 1`, the sum reduces to
 
     ```
-    IAT = -1 + 2 * SUM_{n in N} ac[n],
+    IAT = -1 + 2 * SUM_{n in N} autocorr[n],
```

Comment:
Just wanted to flag that this is really nice & helpful exposition. Thanks!

Reply:
Is it worth explicitly noting that the `-1` is because we double-counted `acor[0]`? It doesn't take much reading between the lines to figure it out, but you never know who's asleep at the wheel.